lib/editor-settings.php

```diff
@@ -77,7 +77,12 @@ function gutenberg_get_common_block_editor_settings() {
  * @return array Filtered settings.
  */
 function gutenberg_extend_post_editor_settings( $settings ) {
+	$image_default_size = get_option( 'image_default_size', 'large' );
+	$image_sizes        = wp_list_pluck( $settings['imageSizes'], 'slug' );
+
+	$settings['imageDefaultSize'] = in_array( $image_default_size, $image_sizes, true ) ? $image_default_size : 'large';
```

Comment:
I see there is a few places where those settings are filtered ...

Maybe, it would be better to keep them next to `imageSizes`:

https://github.com/WordPress/gutenberg/blob/474bb121a9bf8af9db65c623d3eaf796ff01dcf7/lib/editor-settings.php#L49

Reply:
It looks like `gutenberg_get_common_block_editor_settings` isn't used for post editor. Only for site, widgets, and navigation.

I think it makes sense to keep this in the `gutenberg_extend_post_editor_settings` function for now. Saying this based on DocBlocks of both functions.

Reply:
Good catch, it's good to keep as is then. I will investigate separately what's the role of `gutenberg_get_common_block_editor_settings`. FSE and Global Styles add a ton of complexity in the plugin on the PHP side until they are moved to WordPress core.